cmd/ci-images-mirror/main.go

```diff
@@ -320,6 +338,69 @@ func main() {
 	logrus.Info("Process ended gracefully")
 }
 
+func validateConfig(config *quayiociimagesdistributor.CIImagesMirrorConfig, registryConfig string, ciOperatorconfigPath string) error {
+	r, err := regexp.Compile("docker.io")
+	if err != nil {
+		return fmt.Errorf("failed to compile regex: %w", err)
+	}
+	for nsWithName, source := range config.SupplementalCIImages {
+		if source.Image != "" {
+			if match := r.MatchString(source.Image); match {
+				return fmt.Errorf("source image %s is from docker.io", source.Image)
+			}
+			if !isAccessible(source.Image, registryConfig) {
+				return fmt.Errorf("source image %s is not accessible", source.Image)
+			}
+		}
+		if isPromoted(nsWithName, ciOperatorconfigPath) {
+			return fmt.Errorf("target %s would overwrite a promoted tag", nsWithName)
+		}
+		if isDuplicated(nsWithName, config) {
+			return fmt.Errorf("target %s is duplicated", nsWithName)
+		}
+	}
+	return nil
+}
+
+func isPromoted(sourceISTagRef string, ciOperatorconfigPath string) bool {
+	abs, err := filepath.Abs(ciOperatorconfigPath)
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to determine absolute CI Operator configuration path")
+	}
+	err = config.OperateOnCIOperatorConfigDir(abs, func(cfg *api.ReleaseBuildConfiguration, metadata *config.Info) error {
+		for _, isTagRef := range release.PromotedTags(cfg) {
+			if sourceISTagRef == isTagRef.ISTagName() {
+				return fmt.Errorf("source image %s is promoted", sourceISTagRef)
+			}
+		}
+		return nil
+	})
+	return err != nil
+}
+
+func isDuplicated(nsWithName string, config *quayiociimagesdistributor.CIImagesMirrorConfig) bool {
+	var name_count int
+	for name := range config.SupplementalCIImages {
+		if name == nsWithName {
+			name_count++
+		}
+	}
+	return name_count > 1
+}
+
+func isAccessible(image string, registryConfig string) bool {
```

Comment:
This would check all the images in the config.
Ideally, we check only newly added images.